bindings/java/src/org/sleuthkit/datamodel/CaseDbAccessManager.java

```diff
@@ -144,6 +143,78 @@ public boolean columnExists(String tableName, String columnName, CaseDbTransacti
         return columnExists;
     }
 	
+	/**
+	 * Checks if a table exists in the case database.
+	 *
+	 * @param tableName name of the table to check
+	 * 
+	 * @return true if the table already exists, false otherwise
+	 * @throws TskCoreException 
+	 */
+	public boolean tableExists(String tableName) throws TskCoreException {
+		
+		boolean doesTableExist = false;
+		CaseDbTransaction localTrans = tskDB.beginTransaction();
+        try {
+			doesTableExist = tableExists(tableName, localTrans);
+			localTrans.commit();
+			localTrans = null;
+        } 
+		finally {
+			if (null != localTrans) {
+				try {
+					localTrans.rollback();
+				} catch (TskCoreException ex) {
+					logger.log(Level.SEVERE, "Failed to rollback transaction after exception", ex); //NON-NLS
+				}
+			}
+		}
+		
+        return doesTableExist;
+    }
+	
+	/**
+	 * Checks if a table exists in the case database.
+	 *
+	 * @param tableName name of the table to check
+	 * @param transaction transaction 
+	 * 
+	 * @return true if the table already exists, false otherwise
+	 * @throws TskCoreException 
+	 */
+	public boolean tableExists(String tableName, CaseDbTransaction transaction) throws TskCoreException {
+		
+		boolean tableExists = false;
+        Statement statement = null;
+        try {
+			CaseDbConnection connection = transaction.getConnection();
+			statement = connection.createStatement();
+			if (DbType.SQLITE == tskDB.getDatabaseType()) {
+				ResultSet tableQueryResults = statement.executeQuery("SELECT name FROM sqlite_master WHERE type='table'");  //NON-NLS
+				while (tableQueryResults.next()) {
+					if (tableQueryResults.getString("name").equalsIgnoreCase(tableName)) { //NON-NLS
+						tableExists = true;
+						break;
+					}
+				}
+				tableQueryResults.close();
+			}
+			else {
+				String tableInfoQueryTemplate = "SELECT EXISTS (SELECT 1 FROM information_schema.tables WHERE table_name='%s')";  //NON-NLS	
+				ResultSet resultSet = statement.executeQuery(String.format(tableInfoQueryTemplate, tableName.toLowerCase()));
+				if (resultSet.next()) {
```

Comment:
The SELECT EXISTS I don't think really buys us anything, since if the result set is empty it doesn't exist. So I think we could drop the outer query and just say if (resultSet.next()){ tableExists = true; } which would be a check that worked with both this and the proposed query above.

But I am perhaps missing some detail about the queries that makes this a better solution.